### PR TITLE
Return an error when getting coordinates for point at infinity

### LINKED ISSUE #17: Getting cartesian coordinates does not check representation
The `XY` method on the curve point type returns the cartesian coordinates of the point, which is represented in the code in jacobian coordinates. However, this method does not do any checks on the receiver, which can cause problems in the following ways:
- When the receiver is the point at infinity, the result will be meaningless as the point at infinity does not have valid x and y coordinates.
- When the `z` coordinate of the jacobian coordinates is zero, this could potentially cause a runtime exception depending on the implementation of field inversion. At the very least, the zero value, which has no defined multiplicative inverse, will be inverted in the code potentially causing unexpected behaviour.

### PR DESCRIPTION
**Pull request by ross-pure**
This PR fixes #17.

### CODE REVIEW

File: point_test.go [hunk 1]
```diff
@@ -105,7 +117,7 @@ var _ = Describe("Point", func() {
 
 		for i := 0; i < trials; i++ {
 			a = RandomPoint()
-			axf, ayf = a.XY()
+			axf, ayf, _ = a.XY()
```
Comment (loongy) [inline point_test.go]: Consider explicitly confirming that there is no error.

File: point_test.go [hunk 2]
```diff
@@ -160,8 +172,8 @@ var _ = Describe("Point", func() {
 
 		for i := 0; i < trials; i++ {
 			a, b = RandomPoint(), RandomPoint()
-			axf, ayf = a.XY()
-			bxf, byf = b.XY()
+			axf, ayf, _ = a.XY()
+			bxf, byf, _ = b.XY()
```
Comment (loongy) [inline point_test.go]: Consider explicitly confirming that there is no error.

**Review by loongy — CHANGES_REQUESTED** [review]

File: point_test.go (hunk 1 above)
Comment (ross-pure) [reply point_test.go]: Fixed in bf8a281.

File: point_test.go (hunk 2 above)
Comment (ross-pure) [reply point_test.go]: Fixed in bf8a281.

**Pull request merged**